javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/matcher/ClassLoaderHasClassesNamedMatcher.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.extension.matcher;
+
+import io.opentelemetry.instrumentation.api.caching.Cache;
+import io.opentelemetry.javaagent.instrumentation.api.internal.ClassLoaderMatcherCacheHolder;
+import io.opentelemetry.javaagent.instrumentation.api.internal.InClassLoaderMatcher;
+import net.bytebuddy.matcher.ElementMatcher;
+
+class ClassLoaderHasClassesNamedMatcher extends ElementMatcher.Junction.AbstractBase<ClassLoader> {
+
+  private final Cache<ClassLoader, Boolean> cache =
+      Cache.newBuilder().setWeakKeys().setMaximumSize(25).build();
+
+  private final String[] resources;
+
+  ClassLoaderHasClassesNamedMatcher(String... classNames) {
+    resources = classNames;
+    for (int i = 0; i < resources.length; i++) {
+      resources[i] = resources[i].replace(".", "/") + ".class";
```

Comment:
There is `io.opentelemetry.javaagent.tooling.Utils#getResourceName` if you wish

Reply:
`Utils` is in `javaagent-tooling`, it's not accessible here.